packages/bee-q/src/components/panel/bq-panel.tsx

```diff
@@ -0,0 +1,147 @@
+import { h, Component, Prop, Element, Method, Watch, EventEmitter, Event } from '@stencil/core';
+
+import { FloatingUIPlacement } from '../../services/interfaces';
+import { FloatingUI } from '../../services/libraries';
+
+/**
+ * @part base - The `div` element which acts as a container for the panel content
+ */
+@Component({
+  tag: 'bq-panel',
+  styleUrl: './scss/bq-panel.scss',
+  shadow: true,
+})
+export class BqPanel {
+  // Own Properties
+  // ====================
+
+  private panel: HTMLElement;
+  private floatingUI: FloatingUI;
+  private trigger: HTMLElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqPanelElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** Distance (px) between the panel and the trigger element. */
+  @Prop({ reflect: true }) distance?: number = 0;
+
+  /** Position of the panel */
+  @Prop({ reflect: true }) placement?: FloatingUIPlacement = 'bottom';
+
+  /** If true, panel is visible.
+   * You can toggle this attribute to show/hide the panel.
+   */
+  @Prop({ reflect: true }) open?: boolean = false;
+
+  /** If true, the scrollbar is visible.
+   * You can toggle this attribute to show/hide the scrollbar.
+   */
+  @Prop({ reflect: true }) scrollbar?: boolean = false;
+
+  // Prop lifecycle events
+  // =======================
+
+  @Watch('open')
+  handleOpenProp() {
+    this.bqPanelVisibility.emit(this.open);
+  }
+
+  @Watch('distance')
+  @Watch('placement')
+  onPropChange() {
+    this.floatingUI.init({
+      placement: this.placement,
+      distance: this.distance,
+      sameWidth: false,
+      strategy: 'fixed',
+      skidding: 0,
+    });
+  }
```

Comment:
I'm not truly convinced that this logic should belong to the panel. My understanding is that the panel should only handle the styles (shadow, stroke, spacing) and the click/select of the options it has. It should be up to the dropdown itself how to handle showing/hiding the panel, distance, and so on 😉

Reply:
I think that the panel should handle the init of the panel, in our case `initFloatingUI()`, `toggle` panel, and `destroy` panel. Even though these methods are part of the `bq-panel`, they are called from the parent comp, so the control is at the parent level, `bq-dropdown`.
The `bq-panel` component should only handle the styles, as you mentioned, and the create, update floatingUI, "dumb" component. The handle of the selected option is at the `bq-dropdown` level.
Also, in the case of `bq-select`, I think this behavior should follow the same principle. Otherwise we would have to repeat the same code for init, update, destroy.
`bq-select` would be the parent component and have access to `bq-panel` public methods: `setTriggerElement()`, `togglePanel()`.
What do you think?

Reply:
You are ending up duplicating properties to pass from the dropdown parent to the panel component, which I don't think to be ideal, but it could be that maybe I'm missing something. I'll play a bit with your solution in y local env 😉

Reply:
After playing a bit, I agree that we can keep some logic in the `bq-panel`, but there is room for some improvements. I'll provide the relevant changes proposal in the block code related 😉